src/server/pps/server/monitor.go

```diff
@@ -338,6 +342,48 @@ func (m *ppsMaster) monitorPipeline(pachClient *client.APIClient, pipelineInfo *
 			}, backoff.NewInfiniteBackOff(),
 				backoff.NotifyCtx(pachClient.Ctx(), "monitorPipeline for "+pipeline))
 		})
+		if pipelineInfo.ParallelismSpec != nil && pipelineInfo.ParallelismSpec.Constant > 1 && pipelineInfo.Autoscaling {
+			eg.Go(func() error {
+				return backoff.RetryNotify(func() error {
+					worker := work.NewWorker(
+						m.a.env.GetEtcdClient(),
+						m.a.etcdPrefix,
+						driver.WorkNamespace(pipelineInfo),
+					)
+					for {
+						unclaimedTasks, err := worker.UnclaimedTasks(pachClient.Ctx())
+						if err != nil {
+							return err
+						}
+						if unclaimedTasks > 0 {
+							kubeClient := m.a.env.GetKubeClient()
+							namespace := m.a.namespace
+							rc := kubeClient.CoreV1().ReplicationControllers(namespace)
+							scale, err := rc.GetScale(pipelineInfo.WorkerRc, metav1.GetOptions{})
```

Comment:
I think the main risk here is that this code could race with the pipeline controller, but even if it does get the RC into a bad state (e.g. scaling up when a pipeline is supposed to be in standby) the pipeline poller will eventually come along and fix things (fixing weird states due to races is basically the purpose of the pipeline poller).

I think this would all ideally happen in the pipeline controller if possible (to avoid having two concurrent goroutines modifying the RC. It might make sense to add a TODO to that effect?), but I don't think the code will break or anything if we postpone that (and even if we did make that change and all writes came from the same goro, there's no way to completely prevent racing writes to the RC without a write fence). We might just have the occasional pipeline take a long time to go into standby or something.

Reply:
Yeah I had a similar worry here. Figuring out how to propagate this info back to the controller seemed a little tricky. I think I agree that the worst thing I could see happening here is a pipeline being slow to go into standby, I'll add the TODO though.